tools/clean/provider/gcp.go

```diff
@@ -100,6 +103,73 @@ func (gcp *GCP) DeletePrivateEndpoint(ctx context.Context, groupName, sAttachmen
 	return nil
 }
 
+func (gcp *GCP) DeleteOrphanPrivateEndpoints(ctx context.Context, lifetime int, region string, subnet string) error {
+	addresses := gcp.addressClient.List(ctx, &computepb.ListAddressesRequest{
+		Project: gcp.projectID,
+		Region:  region,
+	})
+	for {
+		addr, err := addresses.Next()
+		if errors.Is(err, iterator.Done) {
+			break
+		}
+		if err != nil {
+			return err
```

Comment:
wrap using `fmt.Errorf("error while iterating addresses in project %v region $v: %w", gcp.projectID, region, err)